lockbox-ios/Action/DataStoreAction.swift

```diff
@@ -168,8 +167,8 @@ class DataStoreActionHandler: NSObject, ActionHandler {
 
 // javascript interaction
 extension DataStoreActionHandler {
-    private func _open() -> Observable<Any> {
-        return self.webView.evaluateJavaScript("var \(self.dataStoreName!);swiftOpen().then(function (datastore) {\(self.dataStoreName!) = datastore;});")
+    private func _open() -> Observable<Void> {
+        return self.webView.evaluateJavaScript("var \(self.dataStoreName);swiftOpen().then(function (datastore) {\(self.dataStoreName) = datastore;});")
```

Comment:
If we close (or error) and the reopen again, what happens? 

```js
const theDataStore; swiftOpen().then((dataStore) => theDataStore = dataStore);
```

1. Does `theDataStore` get shadowed, or overwritten or is there an error? Do you just dispose of the webView and start again? 

2. Also, what happens if your datastore is called `dataStore`, that conflicts with your argument name in the `then` block? You should probably either make this difficult to do or leave a comment in the constructor.

Edit: I just realized I typed new style JS. 

More importantly, both of these will cause silent failures. (1) requires some thought about lifecycles, (2) can be fixed with a check in the constructor.